src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -172,15 +174,21 @@ public boolean isStrict()
      */
     public boolean isMutableIntersection()
     {
-        int count = 0;
+        Set<ColumnMetadata> columns = null;
         for (Expression e : expressions)
         {
             if (e.column.isStatic() && expressions.size() > 1)
                 return true;
 
             if (!e.column.isPrimaryKeyColumn())
-                if (++count > 1)
+            {
+                if (columns == null)
+                    columns = new HashSet<>(expressions.size());
```

Comment:
This will be slightly faster if you can use the unique ID of the column because it won't have to hash and equals every single field in column metadata.

Reply:
Went with `IntHashSet` and the unique ID